torch/distributed/fsdp/_common_utils.py

```diff
@@ -184,6 +186,22 @@ def _is_fsdp_flattened(tensor: torch.Tensor) -> bool:
     return getattr(tensor, FSDP_FLATTENED, False)
 
 
+def _named_parameters_with_duplicates(
+    module: nn.Module, **kwargs: Any
+) -> List[Tuple[str, nn.Parameter]]:
+    """
+    This API is required as some modules overwrite `named_parameters()` but do not support
+    `remove_duplicate`.
+    """
+    kwargs["remove_duplicate"] = False
+    try:
+        ret = list(module.named_parameters(**kwargs))
+    except Exception as e:
```

Comment:
Should we only catch the expected exception for when `remove_duplicate` is not supported as an argument instead of catching general `Exception`? It might not make a difference in practice due to how this `except` branch is written, but it might be a good practice.

Reply:
Yes, it does not matter as the exception will be raised again if the error is not what we expect. But since there is a concern about catch all issue, I'll change it to a more specific exception catching.